rspotify-http/src/ureq.rs

```diff
@@ -41,21 +51,28 @@ impl UreqClient {
         }
 
         log::info!("Making request {:?}", request);
+        // Converting errors from ureq into our custom error types
         match send_request(request) {
-            // Successful request
             Ok(response) => response.into_string().map_err(Into::into),
-            // HTTP status error
-            Err(ureq::Error::Status(_, response)) => Err(HttpError::from_response(response)),
-            // Some kind of IO/transport error
-            Err(err) => Err(HttpError::Request(err.to_string())),
+            Err(err) => match err {
+                ureq::Error::Status(_, response) => Err(UreqError::StatusCode(response)),
+                ureq::Error::Transport(transport) => Err(UreqError::Transport(transport)),
```

Comment:
It seems the construction of `UreqError` is not exhaustive, we miss the `UreqError::Io` error type?

Reply:
It's actually used in line 46 with Into::into